rest-server/src/models/job.js

```diff
@@ -138,18 +138,24 @@ class Job {
     }
     userModel.checkUserVc(data.userName, data.virtualCluster, (error, result) => {
       if (!error) {
-        this._prepareJobContext(name, data, (error, result) => {
+        this._initializeJobContextRootFolders((error, result) => {
```

Comment:
do not need to mkdir on every job submission request, only initialize those dirs at the beginning

Reply:
Yep.

Reply:
@abuccts : I think the current implementation, in which all HDFS operations happen when and only when a job is submitted, is easier to maintain. In addition, mkdir operation is light weighted; it will introduce very little overhead to HDFS.